src/db/mysql/models.rs

```diff
@@ -0,0 +1,352 @@
+};
+use diesel::{sql_query, QueryDsl, RunQueryDsl};
+#[cfg(test)]
+use diesel_logger::LoggingConnection;
+use futures::future;
+
+use super::schema::{bso, collections, user_collections};
+#[cfg(test)]
+use super::test::TestTransactionCustomizer;
+use db::{
+    error::DbError, get_std_collection_id, get_std_collection_name, params, results,
+    util::ms_since_epoch, Db, DbFuture, Sorting,
+};
+use settings::Settings;
+
+embed_migrations!();
+
+pub type Result<T> = std::result::Result<T, DbError>;
+
+// The ttl to use for rows that are never supposed to expire (in seconds)
+pub const DEFAULT_BSO_TTL: u32 = 2100000000;
+
+no_arg_sql_function!(last_insert_id, Integer);
+
+/// Run the diesel embedded migrations
+///
+/// Mysql DDL statements implicitly commit which could disrupt MysqlPool's
+/// begin_test_transaction during tests. So this runs on its own separate conn.
+pub fn run_embedded_migrations(settings: &Settings) -> Result<()> {
+    let conn = MysqlConnection::establish(&settings.database_url).unwrap();
+    Ok(embedded_migrations::run(&conn)?)
+}
+
+pub struct MysqlDbPool {
+    pool: Pool<ConnectionManager<MysqlConnection>>,
+}
+
+// XXX: to become a db::DbPool trait
+impl MysqlDbPool {
+    pub fn new(settings: &Settings) -> Result<Self> {
+        let manager = ConnectionManager::<MysqlConnection>::new(settings.database_url.as_ref());
+        let mut builder = Pool::builder().max_size(settings.database_pool_max_size.unwrap_or(10));
+        #[cfg(test)]
+        {
+            if settings.database_use_test_transactions {
+                builder = builder.connection_customizer(Box::new(TestTransactionCustomizer));
+            }
+        }
+        Ok(Self {
+            pool: builder.build(manager)?,
+        })
+    }
+
+    pub fn get(&self) -> Result<MysqlDb> {
+        Ok(MysqlDb {
+            #[cfg(not(test))]
+            conn: self.pool.get()?,
+            #[cfg(test)]
+            conn: LoggingConnection::new(self.pool.get()?),
+        })
+    }
+}
+
+pub struct MysqlDb {
+    #[cfg(not(test))]
+    pub(super) conn: PooledConnection<ConnectionManager<MysqlConnection>>,
+    #[cfg(test)]
+    pub(super) conn: LoggingConnection<PooledConnection<ConnectionManager<MysqlConnection>>>,
+}
+
+impl MysqlDb {
+    pub fn get_collection_id_sync(
+        &self,
+        params: &params::GetCollectionId,
+    ) -> Result<results::GetCollectionId> {
+        let id = if let Some(id) = get_std_collection_id(params) {
+            id
+        } else {
+            sql_query("SELECT id FROM collections WHERE name = ?")
+                .bind::<Text, _>(params)
+                .get_result::<IdResult>(&self.conn)?
+                .id
+        };
+        Ok(id)
+    }
+
+    pub fn get_collections_sync(
+        &self,
+        params: &params::GetCollections,
+    ) -> Result<results::GetCollections> {
+        sql_query("SELECT id, modified FROM collections WHERE user_id = ?")
```

Comment:
ISTM this should be selecting from `user_collections` rather than `collections`.  You could get the names by joining onto the `collections` table, although that may be less efficient when it's only the standard collections for which you already have the names in memory.